src/utils/components/FolderSelect/utils/validation.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+import { t } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { MenuToggleProps } from '@patternfly/react-core';
+
+const isValidFolderNameRegex = /^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$/;
+const startsCorrectlyRegex = /(^$)|(^[A-Za-z0-9])/;
+
+const findInvalidCharacter = (input: string) => {
+  return input.split('').find((char) => /[^-A-Za-z0-9_.]/.test(char));
+};
+
+export const getCreationNotAllowedMessage = (filterValue: string) => {
+  const invalidCharacter = findInvalidCharacter(filterValue);
+
+  if (invalidCharacter) {
+    if (invalidCharacter.trim() === '') {
+      return <>{t('Spaces are not allowed')}</>;
+    }
+
+    return (
+      <>
+        {t('Invalid character')}: {invalidCharacter}
+      </>
+    );
+  }
```

Comment:
don't like the nested `if`s. Can we do 
```suggestion
  if (invalidCharacter) {
      return   <>{invalidCharacter.trim() === '' ? t('Spaces are not allowed') :  {t('Invalid character')}: {invalidCharacter}}</> : ;
    }
  }
```

or something like that